internal/exporter/prometheus/collector/platform_collector.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/sustainable-computing-io/kepler/internal/platform/redfish"
+)
+
+const (
+	// Prometheus namespace for Kepler platform metrics
+	platformSubsystem = "platform"
+)
+
+// RedfishDataProvider defines the interface for getting platform power data
+type RedfishDataProvider interface {
```

Comment:
should this be in prom collector

Reply:
Yes, I feel this is better since there is not direct dependency on Redfish and the "contract" is enforced by the interface. We can move it the redfish service when such an interface is needed.